tests/all/limits.rs

```diff
@@ -752,6 +752,7 @@ fn custom_limiter_detect_grow_failure() -> Result<()> {
         return Ok(());
     }
     let mut pool = crate::small_pool_config();
+    pool.total_memories(2); // 1 for the test, 1 for the GC heap.
```

Comment:
Note to myself: somehow I think we need to address this before merging. At minimum this should probably have a FIXME (or somewhere else), but changing all embeddings to reserve two slots for memory when basically no embedding using GC right now feels pretty bad as well.

Reply:
Ah, this isn't necessary anymore. Removed.